src/main/java/com/google/swarm/tokenization/common/DLPTransform.java

```diff
@@ -157,6 +159,14 @@ public void processElement(ProcessContext c) throws IOException {
             }
           }
         }
+        else{
+          LOG.info("DLPTransform:DLPInspect: "+fileName+" is an empty file | Size of the file in bytes - "+c.element().getValue().length());
+          c.output(
+                  Util.auditData,
+                  Row.withSchema(Util.bqAuditSchema)
+                          .addValues(fileName, Util.getTimeStamp(),0L, "EMPTY")
+                          .build());
+        }
```

Comment:
For the processElement i think error is thrown and caught in the class after DLP is called. Can we have tuple like this?  This should make sure internal error is not going to crash the pipeline. https://github.com/GoogleCloudPlatform/dlp-dataflow-deidentification/blob/2c9bbd178ccaaa4422f89ba49fd0e8b4a0e4f26b/src/main/java/com/google/swarm/tokenization/S3Import.java#L340

Reply:
Yes I got it. But just for clarification the try block is already handling the exception, right? 
We are just adding the catch block to be sure.

Also, this class is already returning a tuple. I will add this additional TupleTag for the Errors.
Is there a need to log these in our main DLPS3ScannerPipeline.java file separately or can we leave it ?

Reply:
Also should we handle this initialization error as follows  : [https://github.com/GoogleCloudPlatform/dlp-dataflow-deidentification/blob/2c9bbd178ccaaa4422f89ba49fd0e8b4a0e4f26b/src/main/java/com/google/swarm/tokenization/S3Import.java#L333](url)

Reply:
you can leave it. Ideally you will flatten all the errors and write back somewhere. processElement throws the error cand catch clock is to output without crashing the pipeline.